IBAnimatable/TransitionAnimationType.swift

```diff
@@ -35,6 +35,75 @@ public enum TransitionAnimationType {
   }
 }
 
+// MARK: - SystemType Initializers
+// Uses TransitionAnimationType.SystemTransitionType to create TransitionAnimationType
+extension TransitionAnimationType {
+  init(systemType: SystemTransitionType, direction: Direction) {
+    switch systemType {
+    case .cube: self = .systemCube(from: direction)
+    case .flip: self = .flip(from: direction)
+    case .moveIn: self = .systemMoveIn(from: direction)
+    case .push: self = .systemPush(from: direction)
+    case .reveal: self = .systemReveal(from: direction)
+    default: fatalError("SystemTransitionType Invalid")
+    }
+  }
+
+  init(systemType: SystemTransitionType) {
+    switch systemType {
+    case .rotate: self = .systemRotate
+    case .suckEffect: self = .systemSuckEffect
+    case .rippleEffect: self = .systemRippleEffect
+    case .pageCurl: self = .systemPage(type: .curl)
+    case .pageUnCurl: self = .systemPage(type: .unCurl)
+    case .cameraIris: self = .systemCameraIris(hollowState: .none)
+    case .cameraIrisHollowOpen: self = .systemCameraIris(hollowState: .open)
+    case .cameraIrisHollowClose: self = .systemCameraIris(hollowState: .close)
+    default: fatalError("SystemTransitionType Requires Direction")
+    }
+  }
+}
+
+extension TransitionAnimationType {
+
+  /// Reverses the direction of the animation
+  var reversed: TransitionAnimationType {
```

Comment:
`reversed` reverses the direction of a `TransitionAnimationType` which the `SystemTransitionAnimator` has -- very confusing I know. That's why if I was able to decouple or re-think these enums a bit I think it would be very useful. However, that's no easy task because the String `initializer` is the most important part of the library. Must tread slowly and with caution.